tests/vector-input.spec.ts

```diff
@@ -0,0 +1,260 @@
+  expect(zTypeValue).toBe('text')
+
+});
+
+test('Renders three inputs with type text if number type specified', async ({ page }) => {
+  const vectorInputNumType = page.getByTestId('vector-input-type-number')
+
+  await expect(vectorInputNumType).toBeVisible()
+
+  const xTypeValue = await vectorInputNumType.locator('input').nth(0).getAttribute("type")
+  expect(xTypeValue).toBe('text')
+  const yTypeValue = await vectorInputNumType.locator('input').nth(1).getAttribute("type")
+  expect(yTypeValue).toBe('text')
+  const zTypeValue = await vectorInputNumType.locator('input').nth(2).getAttribute("type")
+  expect(zTypeValue).toBe('text')
+
+});
+
+test('Renders three inputs with type number if integer type specified', async ({ page }) => {
+  const vectorInputTypeInteger = page.getByTestId('vector-input-type-integer')
+
+  const xTypeValue  = await vectorInputTypeInteger.locator('input').nth(0).getAttribute("type")
+  expect(xTypeValue ).toBe('number')
+  const yTypeValue = await vectorInputTypeInteger.locator('input').nth(1).getAttribute("type")
+  expect(yTypeValue).toBe('number')
+  const zTypeValue = await vectorInputTypeInteger.locator('input').nth(2).getAttribute("type")
+  expect(zTypeValue).toBe('number')
+
+});
+
+test('When inputs are updated, events fire', async ({ page }) => {
+  const vectorInputOnInput = page.getByTestId('vector-input-on-input')
+
+  await expect(vectorInputOnInput).toBeVisible()
+
+  const xInput = vectorInputOnInput.locator('input').nth(0)
+  const xInputEvent = waitForCustomEventWithParam(page, 'input', 'value')
+  await xInput.focus()
+  await xInput.press('ArrowUp')
+  expect(xInputEvent).toBeTruthy()
+  expect(await getCustomEventParam(page, 'input', 'value')).toBe('1')
+  
+  const yInput = vectorInputOnInput.locator('input').nth(1)
+  const yInputEvent = waitForCustomEventWithParam(page, 'input', 'value')
+  await yInput.focus()
+  await yInput.press('ArrowUp')
+  expect(yInputEvent).toBeTruthy()
+  expect(await getCustomEventParam(page, 'input', 'value')).toBe('1,1')
+
+
+  const zInput = vectorInputOnInput.locator('input').nth(2)
+  const zInputEvent = waitForCustomEventWithParam(page, 'input', 'value')
+  await zInput.focus()
+  await zInput.press('ArrowUp')
+  expect(zInputEvent).toBeTruthy()
+  expect(await getCustomEventParam(page, 'input', 'value')).toBe('1,1,1')
+
+});
+
+test('When label is specified, label renders to the left of inputs', async ({ page }) => {
+  const vectorInputLabelLeft = page.getByTestId('vector-input-label')
+
+  await expect(vectorInputLabelLeft).toBeVisible()
+
+  const labelText = await vectorInputLabelLeft.getByRole('paragraph').textContent()
+  expect(labelText).toBe('Label Left')
+
+  // Confirm label is to the left of x input (x input is to the right of label)
+  const xInputPlaceholder = (await page.locator("input:right-of(:text('Label Left'))").nth(1).getAttribute('placeholder'))
```

Comment:
this is a triumph. we will overcome the position-based locators !

Reply:
:laughing: